Tests/Shared.Specs/FakeClock.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions.Common;
+
+namespace FluentAssertions.Specs
+{
+    /// <summary>
+    /// Implementation of <see cref="IClock"/> for testing purposes only.
+    /// </summary>
+    /// <remarks>
+    /// It allows you to control the "current" time.
+    /// </remarks>
+    internal class FakeClock : IClock
+    {
+        private DateTime fakeNow = DateTime.Now;
+
+        private TaskCompletionSource<bool> delayTask { get; } = new TaskCompletionSource<bool>();
+
+        Task IClock.DelayAsync(TimeSpan delay, CancellationToken cancellationToken)
+        {
+            fakeNow = fakeNow.Add(delay);
```

Comment:
🙃 `DateTime` has nice (in my opinion) overloaded operators such as `+=`, `-` instead of `Add` and `Subtract`.